numbagg/funcs.py

```diff
@@ -1,10 +1,10 @@
 import numpy as np
-from numba import bool_, float32, float64, int64
+from numba import bool_, float32, float64, int32, int64
 
 from .decorators import ndreduce
 
 
-@ndreduce([bool_(float32), bool_(float64)])
+@ndreduce([bool_(int32), bool_(int64), bool_(float32), bool_(float64)])
```

Comment:
Not harm in making all the functions work for int dtypes; though `allnan` presumably isn't applicable for ints?

Reply:
Indeed, this will always return `False`. But it's convenient to have around so downstream uses need fewer special cases.

Note that Numba/LLVM is actually smart enough to avoid the loop entirely for integer dtypes in anynan. Compare:
```
In [26]: x = np.arange(10000000)

In [27]: y = 1.0 * x

In [28]: %timeit numbagg.anynan(x)
1.31 µs ± 77.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [29]: %timeit numbagg.anynan(y)
14.4 ms ± 777 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
That's awesome!